### PR TITLE
[bitnami/external-dns] Add support for externalName service

### PR DESCRIPTION
**Pull request by fevisera**
### Description of the change

Add support for using `externalName` service.

### Benefits

Users can setup ExternalDNS with an ExternalName service.

### Possible drawbacks

N/A

### Applicable issues

- fixes #

### Additional information

[Example](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/externalname.md) of configuring externalName.

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/external-dns/templates/service.yaml [hunk 1]
```diff
@@ -30,6 +30,9 @@ spec:
   {{- if .Values.service.externalIPs }}
   externalIPs: {{ toYaml .Values.service.externalIPs | nindent 4 }}
   {{- end }}
+  {{- if .Values.service.externalName }}
```
Comment (fmulero) [inline bitnami/external-dns/templates/service.yaml:33]: This new value only makes sense when [`service.type=ExternalName`](https://kubernetes.io/docs/concepts/services-networking/service/#externalname), I think we should add a condition like this:
```suggestion
  {{- if and (eq .Values.service.type "ExternalName") .Values.service.externalName }}
```

**Review by fmulero — COMMENTED** [review]
Thanks @fevisera for your contibution.

I left only a little suggestion. Could you give it a glance?

File: bitnami/external-dns/templates/service.yaml (hunk 1 above)
Comment (fevisera) [reply bitnami/external-dns/templates/service.yaml]: Thank you for your review! You are right, I have applied your suggestion. Thanks.

**Review by fmulero — COMMENTED** [review]
Great! LGTM

**Review by fmulero — APPROVED** [review]

**Pull request merged**